db/seeds.d/20-discovery-roles.rb

```diff
@@ -0,0 +1,31 @@
+class DiscoveryPermissions
+  READER = [
+    :view_discovered_hosts,
+    :view_discovery_rules,
+    :view_locations,
+    :view_organizations,
+  ]
+
+  MANAGER = READER + [
+    :assign_organizations,
+    :assign_locations,
+    # discovered_hosts
+    :submit_discovered_hosts,
+    :auto_provision_discovered_hosts,
+    :provision_discovered_hosts,
+    :edit_discovered_hosts,
+    :destroy_discovered_hosts,
+    # discovered_rules
+    :create_discovery_rules,
+    :edit_discovery_rules,
+    :execute_discovery_rules,
+    :destroy_discovery_rules,
+  ]
+end
+
+unless Rails.env.test?
+  Role.without_auditing do
+    create_role('Discovery Reader', DiscoveryPermissions::READER, 0)
+    create_role('Discovery Manager', DiscoveryPermissions::MANAGER, 0)
+  end
+end
```

Comment:
I would guess that (1) and (2) could probably be fixed in Foreman, and I don't know what (3) is.

If you want to deprecate it, then first let's see if the issues can be fixed to avoid disruption, and if not, then you need to be filing a PR against Foreman itself to mark the feature as deprecated.